### ISSUE TITLE
Issue with moons

### ISSUE THREAD

**Issue by Roshex** [SethFusion/US2-to-SE-Converter#2]
Dear Seth,

Just joined Github, so apologies if that's not how people normally leave comments here (not sure how?). Anyway, thanks for this awesome tool! There is however an issue that fails to identify moons that fail the "tag-of-war" definition if their planet orbits a binary. "moons that fail the tag-of-war" also include our own Moon, see here:

https://en.wikipedia.org/wiki/Orbit_of_the_Moon 
under:Path of Earth and Moon around Sun
"The Sun's gravitational effect on the Moon is more than twice that of Earth's on the Moon; consequently, the Moon's trajectory is always convex[24][25] (as seen when looking Sunward at the entire Sun–Earth–Moon system from a great distance outside Earth–Moon solar orbit), and is nowhere concave (from the same perspective) or looped.[23][24][26] That is, the region enclosed by the Moon's orbit of the Sun is a convex set."

https://en.wikipedia.org/wiki/Double_planet
under:Tug-of-war value
"In nearly every one of Asimov's calculations the tug-of-war value was found to be greater than one, so in those cases the Sun loses the tug-of-war with the planets. The one exception was Earth's Moon, where the Sun wins the tug-of-war with a value of 0.46, which means that Earth's hold on the Moon is less than half that of the Sun's. Asimov included this with his other arguments that Earth and the Moon should be considered a binary planet.[5]"

In your code there's a loop that excludes "Moon-like moons" if the star is a binary / barycenter:

https://github.com/SethFusion/US2-to-SE-Converter/blob/6d0ae88b22832193f8d8bec2eff1a98bbcbf6f27/US2%20to%20SE%20Converter/US2%20to%20SE%20Converter.cpp#L263

Not sure if that check is necessary at all since US2 should have separated any "Moon-like moons" from their planets if they strayed too far from it. Alternatively, you'd need some other condition to allow moons that lose the tag-of-war if they co-orbit with another body (which would be their planet) - but that would require a complicated solution... The Moon and the Earth only co-orbit because they formed from the same body, so their orbits just happen to match (in period, ecc, argofperi, etc).

Thanks in advance!

**Comment by SethFusion** [comment]
Roshex, 

Thank you for taking the time to not only use the program I created, but also to offer suggestions about the accuracy of the program itself. I've been pretty open about not being a physicist, or even very knowledgeable about planetary motion to begin with, I'm just a hobbyist. I learned enough to create this program, but it is far from perfect, so I appreciate the input. 

First off, let me apologize in advance. It's been a few months since I worked on this program, and sometimes looking at old code is not so easy. I'll try to remember exactly what I did and why I did it. Also, give me a bit of time to look at all of the issues you posted so I can reply to them correctly and address them if needed. Also note that I probably will not fix these issue right away, but I do intend to come back to this project and upgrade it before the end of the year. I'm working on another project right now that is eating up my time. Once I get back to campus, I'll have access to some old professors that might also be able to help with the math behind this program. In the future, I also want to support converting from SE to US. 

So for the moon issue, I believe you are right. I can't remember exactly why I put this line in, but I think my intention was to determine if a single planet in a binary-planet system was the object's parent, or if the whole binary-planet system was the object's parent. For instance, a binary planet system can have many objects orbiting the barycenter apart from the two main bodies. These would be asteroids and such objects too small to affect the two main bodies in any way. However, it could also be possible that an asteroid could orbit just one of the main bodies in a binary-planet system if the two main bodies were far enough apart. Such a system could look like the Earth-Moon system. If an asteroid were orbiting very close to Earth, it's parent would be Earth, not the Earth-Moon barycenter, and Earth's parent would still be the Earth-Moon barycenter. Anyway, perhaps I need some kind of check here to make sure to only look for planetary barycenters and not stellar barycenters...

>  Alternatively, you'd need some other condition to allow moons that lose the tag-of-war if they co-orbit with another body (which would be their planet) - but that would require a complicated solution... 

So this was a problem back when I first wrote this, but the basic idea was not to calculate anything unless necessary. I actually just assume that if a moon is close enough to a planet (within its sphere of influence), it is probably the child of that planet, and that covers most simple cases. Anything more complicated would have required math that I didn't know. But I didn't know about the tug-of-war equation you linked above, so when I re-write some of this code in the future, that will be super useful.

Again, thanks for the input. Give me some time to look through your other posts.
